src/media/local-stream.ts

```diff
@@ -141,55 +144,80 @@ abstract class _LocalStream extends Stream {
   /**
    * Adds an effect to a local stream.
    *
-   * @param name - The name of the effect.
    * @param effect - The effect to add.
    */
-  async addEffect(name: string, effect: TrackEffect): Promise<void> {
-    // Load the effect
-    this.loadingEffects.set(name, effect);
-    const outputTrack = await effect.load(this.outputTrack);
+  async addEffect(effect: TrackEffect): Promise<void> {
+    // Check if the effect has already been added.
+    if (this.effects.includes(effect)) {
```

Comment:
What does this end up using to compare equality?

Reply:
It just checks to see if the instances match, but I do think explicitly comparing `id` or `kind` would be better, so I can make that change.